spec/requests/signup_spec.rb

```diff
@@ -0,0 +1,103 @@
+require 'jwt'
+require 'rails_helper'
+# rubocop:disable Metrics/BlockLength
+RSpec.describe 'POST /signup', type: :request do
+  # write test for what happens with application/json Accept header
+
+  describe 'when invalid params' do
+    let(:user) { User.first }
+    describe 'missing parameter' do
+      before :all do
+        User.destroy_all
+        headers = {
+          'CONTENT_TYPE' => 'application/vnd.api+json'
+        }
+
+        params = JSON.generate({
+          user: {
+            email: 'testuser@email.com',
+            password: 'password'
+          }
+        })
+
+        post '/signup', headers: headers, params: params
+      end
+
+      it 'response is 400' do
+        expect(response).to have_http_status(400)
+      end
+
+      it 'user is not created' do
+        expect(User.count).to eq(0)
+      end
+    end
+  end
+
+  describe 'when request is valid' do
+    let(:user) { User.first }
+    before :all do
+      # thought there wouldn't be a need for this. Why doesn't the test db reset after each run of the suite?
```

Comment:
Look into this. Perhaps it's the model tests that mean a User is persisted to the db before this file is run?